pushsafer/pushsafer_test.go

```diff
@@ -0,0 +1,101 @@
+package pushsafer
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/variadico/noti"
+)
+
+func TestEnvConfig(t *testing.T) {
+	cases := []struct {
+		env     noti.MockEnv
+		config  configuration
+		wantErr bool
+		err     error
+	}{
+		{
+			env:     noti.MockEnv{},
+			config:  configuration{},
+			wantErr: true,
+			err:     errNoKey,
+		},
+		{
+			env:     noti.MockEnv{keyEnv: "fu"},
+			config:  configuration{privateKey: "fu"},
+			wantErr: false,
+			err:     nil,
+		},
+	}
+
+	for i, c := range cases {
+		config, err := envConfig(c.env)
+		gotErr := (err != nil)
+
+		if gotErr && !c.wantErr {
+			t.Error(i, "unexpected error")
+			t.Error(err)
+		} else if gotErr && c.wantErr {
+			if err != c.err {
+				t.Error(i, "unexpected error")
+				t.Error(err)
+			}
+		}
+
+		if config != c.config {
+			t.Error(i, "unexpected configuration")
+			t.Errorf(" got: %v", config)
+			t.Errorf("want: %v", c.config)
+		}
+	}
+}
+
+func TestNotify(t *testing.T) {
+	n := noti.Params{
+		Title:   "title",
+		Message: "mesg",
+		API:     API,
+		Config:  noti.MockEnv{keyEnv: "fu"},
+	}
+	var mockResp apiResponse
+	var hitServer bool
+
+	ts := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
+		hitServer = true
+
+		if r.Method != "POST" {
+			t.Error("HTTP method should be POST")
+		}
+
+		if r.FormValue("privateKey") == "" {
```

Comment:
this line: https://github.com/variadico/noti/pull/58/files#diff-4dd7e1034427325b97febafe3784a910R56 shows that the api is expecting `"k"` instead of `"privateKey"`. So instead of this, try:
```
if r.FormValue("k") == "" {
```